apps/condo/domains/common/components/AudioPlayer.tsx

```diff
@@ -0,0 +1,155 @@
+
+interface IAudioPlayerProps {
+    src: string
+    trackId: string
+    autoPlay?: boolean
+}
+
+export const AudioPlayer: React.FC<IAudioPlayerProps> = ({ trackId, src, autoPlay }) => {
+    const intl = useIntl()
+    const SpeedMessage = intl.formatMessage({ id: 'ticket.callRecord.speed' })
+
+    const [playing, setPlaying] = useState(false)
+    const [currentTime, setCurrentTime] = useState('00:00')
+    const [totalTime, setTotalTime] = useState('00:00')
+    const [speed, setSpeed] = useState(1)
+    const waveformRef = useRef<HTMLDivElement>(null)
+    const waveform = useRef<WaveSurfer>()
+
+    useEffect(() => {
+        if (typeof window !== 'undefined') {
+            const track: HTMLMediaElement = document.querySelector(`#${trackId}`)
+
+            waveform.current = WaveSurfer.create({
+                container: waveformRef.current,
+                height: 28,
+                progressColor: createGradient(),
+                waveColor: colors.gray[5],
+                responsive: true,
+                barGap: 1,
+                barWidth: 2,
+                barHeight: 28,
+                cursorWidth: 0,
+                hideScrollbar: true,
+            })
+
+            waveform.current.on('ready', () => {
+                setTotalTime(formatTime(waveform.current.getDuration()))
+
+                if (autoPlay) {
+                    waveform.current.play()
+                    setPlaying(true)
+                }
+            })
+
+            waveform.current.on('audioprocess', () => {
+                setCurrentTime(formatTime(waveform.current.getCurrentTime()))
+            })
+
+            waveform.current.on('finish', () => {
+                setPlaying(false)
+            })
+
+            waveform.current.load(track)
+
+            return () => {
+                if (waveform.current) {
+                    waveform.current.destroy()
+                }
+            }
+        }
+    }, [])
+
+    const handlePlay = () => {
+        setPlaying(!playing)
+        waveform.current.playPause()
+    }
+
+    const handleSpeedChange = (value: number) => {
+        setSpeed(value)
+        waveform.current.setPlaybackRate(value)
+    }
+
+    const PlayerIcon = playing ? Pause : Play
+
+    const formatTime = (time: number): string => {
+        const minutes = Math.floor(time / 60).toString().padStart(2, '0')
+        const seconds = Math.floor(time % 60).toString().padStart(2, '0')
+        return `${minutes}:${seconds}`
+    }
+
+    const createGradient = (): CanvasGradient => {
+        const canvas = document.createElement('canvas')
+        const ctx = canvas.getContext('2d')
+        const gradient = ctx.createLinearGradient(0, 0, canvas.width, 0)
+        gradient.addColorStop(0, '#4CD174')
+        gradient.addColorStop(1, '#6DB8F2')
+
+        return gradient
+    }
+
+    return (
+        <Row gutter={PLAYER_ROW_GUTTER} align='middle'>
```

Comment:
Are sure that Row / Col approach will be adaptive? (since all that dynamic spacing size it have)
If adaptive is not the goal - just skip this comment :)